cpp/include/cuvs/neighbors/cagra.h

```diff
@@ -392,6 +392,22 @@ cuvsError_t cuvsCagraSearch(cuvsResources_t res,
                             DLManagedTensor* queries,
                             DLManagedTensor* neighbors,
                             DLManagedTensor* distances);
+/*
+ * @param[in] res cuvsResources_t opaque C handle
+ * @param[in] params cuvsCagraSearchParams_t used to search CAGRA index
+ * @param[in] index cuvsCagraIndex which has been returned by `cuvsCagraBuild`
+ * @param[in] queries DLManagedTensor* queries dataset to search
+ * @param[in] filter Filter
+ * @param[out] neighbors DLManagedTensor* output `k` neighbors for queries
+ * @param[out] distances DLManagedTensor* output `k` distances for queries
+ */
+cuvsError_t cuvsCagraFilteredSearch(cuvsResources_t res,
+                                    cuvsCagraSearchParams_t params,
+                                    cuvsCagraIndex_t index,
+                                    DLManagedTensor* queries,
+                                    DLManagedTensor* neighbors,
+                                    DLManagedTensor* distances,
+                                    DLManagedTensor* filter);
```

Comment:
Instead of having a different method for the filtered search - we are trying to just have one search method, with the 'filter' being an optional argument. 

As an example check out the bfknn code here 

https://github.com/rapidsai/cuvs/blob/5062594138a40231475299c7bac61083b0669fd1/cpp/include/cuvs/neighbors/brute_force.h#L159-L164

Also rather than passing a `DLManagedTensor* filter` object - can we pass the same `cuvsFilter prefilter`  that the bfknn code is doing ? I'd like to keep the API's consistent between the various search algorithms.

Reply:
Good idea, I just modified it to be more similar to the bfknn API. One potential issue is that every search call now needs to have a filter, even if no filter is used (same as the bfknn API).
Keep in mind that bfknn only supports bitmaps, and cagra only supports bitsets as filters.